ciao-cli/main.go

```diff
@@ -1031,7 +1038,12 @@ func main() {
 		if *imageName == "" {
 			fatalf("Missing required -image-name parameter")
 		}
-		createTenantImage(*identityUser, *identityPassword, id, *imageName)
+		if *imageVisibility != "private" && *imageVisibility != "public" {
```

Comment:
Here you want to define e.g. `type visibility string`, implement `func (role *visibility) Set(value string) error` and then call
```
var imageVisibility visibility
flag Var(&imageVisibility, ""image-visibility", "Image visibility. Value can be public or private")
```
The `Set()` method will allow you to define what's acceptable for a visibility value.
The same logic should apply to container and disk formats.